### PR TITLE
Openshift

### PR DESCRIPTION
**Pull request by noseka1**
## Related Issues and Dependencies

…

## This introduces a breaking change

- [ ] Yes
- [ ] No

## This Pull Request implements

Update to Airflow 1.10.12
User login with OpenShift

## Description

Individual commits include the detailed description.

### CODE REVIEW

File: patch/flash_appbuilder.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+diff --git a/flask_appbuilder/security/manager.py b/flask_appbuilder/security/manager.py
+index e2b361c9..a454d8c6 100644
+--- a/flask_appbuilder/security/manager.py
```
Comment (vpavlin) [inline patch/flash_appbuilder.patch]: Should the file name to be `flask_appbuilder.patch` rather than `flash_...`?

Comment (noseka1) [reply patch/flash_appbuilder.patch]: Ooops, indeed. I pushed the fix.

**Comment by vpavlin** [comment]
I tried to deploy this to my cluster and I see 

```
+ update-ca-certificates --localcertsdir /tmp/tmp.9cdYd8zXRM
Updating certificates in /etc/ssl/certs...
6 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
+ /configure_auth.py
Custom webserver configuration written to /usr/local/airflow/webserver_config.py
+ exec airflow webserver
```

In the logs, but when I access the UI route there is not auth happening.

**Comment by vpavlin** [comment]
FYI @noseka1 I used this Cluster spec

```
apiVersion: airflow.apache.org/v1alpha1
kind: AirflowCluster
metadata:
  name: pc-cluster
spec:
  config:
    airflow:
      AUTH_TYPE: openshift
  airflowbase:
    name: pc-base
  dags:
    git:
      branch: master
      once: true
      repo: 'https://github.com/VedantMahabaleshwarkar/airflow-dags'
    subdir: airflow/dags/
  executor: Celery
  ui:
    enableroutes: true
    image: quay.io/opendatahub/docker-airflow
    replicas: 1
    resources: {}
    version: pr-10
  scheduler:
    database: j0tdmp6fb8c8fwuu
    dbuser: vu93bk12r4pnvjlr
    image: quay.io/opendatahub/docker-airflow
    resources: {}
    version: pr-10
  redis:
    image: redis
    resources: {}
    version: '4.0'
  flower:
    enableroutes: true
    image: quay.io/opendatahub/docker-airflow
    replicas: 1
    resources: {}
    version: pr-10
  worker:
    forceroot: 'false'
    image: quay.io/opendatahub/docker-airflow
    replicas: 2
    resources: {}
    version: pr-10

```

**Comment by noseka1** [comment]
@vpavlin Thanks for checking it out. An updated version of the Airflow operator is needed to make the Login with OpenShift work. The updated operator is available here: https://github.com/noseka1/airflow-on-k8s-operator/commits/openshift-3 I included the Cluster spec examples in the comment: https://github.com/noseka1/airflow-on-k8s-operator/commit/3c4a27dc2be2f984c50b257a9c7a05ca318d75c7

**Comment by vpavlin** [comment]
So what is the plan for that branch? Submit a PR after this is merged?

**Comment by noseka1** [comment]
The plan back in July was to use my openshift-2 or now openshift-3 branch as a replacement for the openshift branch in opendatahub-io/airflow-on-k8s-operator. If we want to continue with this plan, we would need to replay the commit https://github.com/opendatahub-io/airflow-on-k8s-operator/commit/fbfc7e67f0f44eb31f2ba33be6f2d1caccc37d69 on top of openshift-3 branch.

Alternative plan is for me to rebase the openshift-3 branch on top of the current openshift branch in opendatahub-io/airflow-on-k8s-operator. I would revert the commits made in the openshift branch in April though, as they are IMO of not a good quality. I could then create a pull request.

Let me know which of the two alternatives above you prefer. Thanks.

**Comment by vpavlin** [comment]
I am fine with either. The changes from April were made by an internet and it was his first experience with golang. So as long as we keep the features, I am fine dropping those. Whatever is better/easier for you.

**Comment by noseka1** [comment]
Let's continue with the original plan then. I think these are the next steps:

1. Replace the openshift branch in https://github.com/opendatahub-io/airflow-on-k8s-operator with the openshift-3 branch from https://github.com/noseka1/airflow-on-k8s-operator. openshift-3 branch contains all the features from the original openshift branch.
2. Replay the commit https://github.com/opendatahub-io/airflow-on-k8s-operator/commit/fbfc7e67f0f44eb31f2ba33be6f2d1caccc37d69 on top of the updated openshift branch.

The above work needs to be done in the https://github.com/opendatahub-io/airflow-on-k8s-operator repo, where I don't have access.

**Comment by vpavlin** [comment]
I think that with your involvment you more than deserve to have the write access  to the airflow ODH repositories - I've sent the invites. 

It would be great if we can keep the PR workflow with me, or someone else from ODH, reviewing the PRs

**Comment by noseka1** [comment]
@vpavlin Thanks for granting me write access to the Airflow ODH repos.

I cherry-picked the commit https://github.com/opendatahub-io/airflow-on-k8s-operator/commit/fbfc7e67f0f44eb31f2ba33be6f2d1caccc37d69 and put it on top of https://github.com/noseka1/airflow-on-k8s-operator/tree/openshift-3 Next steps I am about to do are:
1. Push my openshift-3 branch to https://github.com/opendatahub-io/airflow-on-k8s-operator
2. Wait for the CI to complete (I assume the CI will run after I push the openshift-3 branch)
3. Rename openshift branch to openshift-obsolete
4. Rename openshift-3 branch to openshift

Can you confirm if this is correct? If so, should I proceed?

**Comment by vpavlin** [comment]
Hi @noseka1, sorry for late reply, completely missed this one:( 

Yes, I think the list of steps it correct. One note - the "CI" just builds the image, does not run any tests, so if you'd like me to, I can run a build and test it manually to make sure everything works as expected.

Otherwise, it sounds good.

**Comment by noseka1** [comment]
Hi @vpavlin,

I completed the step 1 and 2 as described above. I didn't really see the CI running.

It appears to me that before proceeding with steps 3 and 4, we should actually:

a) Complete the review and merge this pull request
b} Build the image docker-airflow:1.10.12 and push it to https://quay.io/repository/opendatahub/docker-airflow

The latest Airflow operator deploys the docker-airflow:1.10.12 image. I think the steps a) and b) would be best done by you.

**Comment by vpavlin** [comment]
LGTM

**Pull request merged**